packages/go_router/lib/src/delegate.dart

```diff
@@ -148,9 +148,18 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
     return navigatorKey.currentState?.canPop() ?? false;
   }
 
+  void _debugAssertMatchListNotEmpty() {
+    assert(
+      _matchList.isNotEmpty,
+      'You have popped the last page off of the stack,'
+      ' there are no pages left to show',
+    );
+  }
+
   /// Pop the top page off the GoRouter's page stack.
   void pop() {
     _matchList.pop();
+    _debugAssertMatchListNotEmpty();
```

Comment:
you can also wrap this in an assert so that release mode will not even waste time calling into an empty function

Reply:
Is https://github.com/flutter/packages/pull/2747/commits/8e8d1873c89807b9a87f21c4b6b6233c0c7aded8 what you mean?

Reply:
yes